tests/testSBMLSuite.py

```diff
@@ -106,13 +110,12 @@ def verify_results(settings, rdata, expected, wrapper,
         settings['concentration'].replace(' ', '').replace('\n', '').split(',')
         if s
     ]
-    # The rate rules condition here may be unnecessary/better implemented
-    # elsewhere.
+    # We only need to convert species that have only substance units and that
+    # are targets of rate rules
     concentration_species = [
-        species for species in requested_concentrations
-        if wrapper.species_has_only_substance_units[
-               wrapper.species_index[species]
-        ] and species in [str(s) for s in wrapper.species_rate_rules]
+        str(species_id)
+        for species_id, species in wrapper.symbols[SymbolId.SPECIES].items()
+        if str(species_id) in requested_concentrations and species['amount']
     ]
```

Comment:
I guess entries in `wrapper.symbols[SymbolId.SPECIES]`are always targets of rate rules? If not, the comment still mentions rate rules, but no code related to rate rules seems to be here.